extension/src/background/messageListener/freighterApiMessageListener.ts

```diff
@@ -294,13 +294,20 @@ export const freighterApiMessageListener = (
     });
   };
 
-  const requestUserInfo = () => {
+  const requestUserInfo = async () => {
     const publicKey = publicKeySelector(sessionStore.getState());
+    const isAllowed = await isSenderAllowed({ sender });
```

Comment:
make sure to check allowlist before transmitting public key

Reply:
ah sorry I didn't see this either